pulp_container/app/serializers.py

```diff
@@ -35,7 +35,7 @@
 
 VALID_SIGNATURE_NAME_REGEX = r"^sha256:[0-9a-f]{64}@[0-9a-f]{32}$"
 VALID_TAG_REGEX = r"^[A-Za-z0-9][A-Za-z0-9._-]*$"
-VALID_BASE_PATH_REGEX_COMPILED = re.compile(r"^[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?$")
+VALID_BASE_PATH_REGEX_COMPILED = re.compile(r"^[a-z0-9]+((?:[._]|__|[-]*)[a-z0-9])*$")
```

Comment:
We could make the outer capture group anonymous too:
```suggestion
VALID_BASE_PATH_REGEX_COMPILED = re.compile(r"^[a-z0-9]+(?:(?:[._]|__|[-]*)[a-z0-9])*$")
```

Also possible, we could add the slashes here and not split later:
`r"^[a-z0-9]+(?:(?:[._]|__|[-]*)[a-z0-9])(?:/[a-z0-9]+(?:(?:[._]|__|[-]*)[a-z0-9]))*$"`